File: Corrections/java/Neufplate/src/main/java/fr/neufplate/PriceCalculator/PriceDecorator.java
package fr.neufplate.PriceCalculator;

public class PriceDecorator extends Price {
    Price price;

    public PriceDecorator(Price price) {
        this.price = price;
    }

    @Override
    public double calculate() {
        return this.price.calculate();
    }

}

File: Corrections/php/Neufplate/src/PriceCalculator/PriceDecorator.php
<?php

namespace App\PriceCalculator;

class PriceDecorator extends Price
{

    public Price $price;

    public function __construct(Price $price)
    {
        $this->price = $price;
    }

    public function calculate(): float
    {
        return $this->price->calculate();
    }
}